src/builders/utils/ios/adhocBuild.ts

```diff
@@ -21,10 +20,18 @@ async function prepareAdHocBuildCredentials(job: IJob) {
   }
 
   const { bundleIdentifier } = job.config;
-  const { certP12, certPassword, teamId, appleSession, udids } = job.credentials;
+  const {
+    certP12,
+    certPassword,
+    teamId,
+    appleSession,
+    udids,
+    provisioningProfileId,
+  } = job.credentials;
 
   const certSerialNumber = IosCodeSigning.findP12CertSerialNumber(certP12, certPassword);
   const args = [
+    ...(provisioningProfileId ? ['--profile-id', provisioningProfileId] : []),
```

Comment:
Please don't mix positional parameters with named ones. All of the parameters passed to this script are positional so if something is optional is the best to put it as the last parameter (and remove the naming).